setup.py

```diff
@@ -132,7 +131,16 @@
       ],
       scripts=["bin/allennlp"],
       setup_requires=setup_requirements,
-      tests_require=['pytest', 'flaky', 'responses>=0.7', 'jupyter'],
+      tests_require=[
```

Comment:
If my understanding is correct, these packages listed under `tests_require` should correspond to packages listed in the section "TESTING-RELATED PACKAGES" in `requirements.txt`

Reply:
👍 you're correct

Reply:
accidentally deleted my original comment, sorry.

I think it's a bit more nuanced than that --- a lot of the things in the `TESTING-RELATED PACKAGES` aren't actually needed to run the tests (e.g., codecov). They're just used in CI. I think the idea was to only put things that are actually needed to run the tests in `tests_require` (so you should be able to remove `mypy`, `codecov`, `pytest-cov`, `pylint`, `coverage` (I'll have to check on `aiohttp` and `moto`)

Reply:
So in that case, would `tests_require` correspond to `ESSENTIAL TESTING-RELATED PACKAGES`?

Reply:
I think we can actually remove `aiohttp`, since it was used to run tests for sanic. however, I think we've switched over to flask (@joelgrus / @schmmd can confirm).

Reply:
Yes, that's fair. though `conllu` should be in  `#### ESSENTIAL LIBRARIES FOR MAIN FUNCTIONALITY ####`, would you mind moving it?

Reply:
Got it!

Reply:
@nelson-liu I have no idea by now--I'd cut a PR and see what CI says.